python/pyiceberg/table/partitioning.py

```diff
@@ -82,19 +82,16 @@ class PartitionSpec(IcebergBaseModel):
         fields(List[PartitionField): list of partition fields to produce partition values
     """
 
-    spec_id: int = Field(alias="spec-id")
-    fields: Tuple[PartitionField, ...] = Field(default_factory=tuple)
+    spec_id: int = Field(alias="spec-id", default=INITIAL_PARTITION_SPEC_ID)
```

Comment:
I think I'd prefer to handle ID assignment manually rather than defaulting. Defaulting seems to bring in complexity because if we forget to pass along an ID somewhere, it would cause problems.

Reply:
I feel that we don't should really expose this to the user. For example, when we create a new table, we re-assign the IDs anyway (using the assign fresh IDs logic).
If we follow the Java API, and we have something similar to `updateSpec`: https://github.com/apache/iceberg/blob/master/api/src/main/java/org/apache/iceberg/Table.java#L165-L171 Then we can just take the next ID. What do you think of this?

Reply:
That sounds reasonable to me.

Reply:
This will definitely involve a lot of testing 👍🏻